src/gui/render.cpp

```diff
@@ -615,26 +618,35 @@ void RENDER_SetForceUpdate(bool f) {
 }
 
 #if C_OPENGL
-static bool RENDER_GetShader(std::string& shader_path, char *old_src) {
+static bool RENDER_GetShader(std::string &shader_path, char *old_src)
+{
+	const std::unordered_map<std::string, const char *> builtin_shaders = {
+	        {"advinterp2x", advinterp2x_glsl},
+	        {"advinterp3x", advinterp3x_glsl},
+	        {"advmame2x", advmame2x_glsl},
+	        {"advmame3x", advmame3x_glsl},
+	        {"crt-easymode-flat", crt_easymode_tweaked_glsl},
+	        {"crt-fakelottes-flat", crt_fakelottes_tweaked_glsl},
+	        {"rgb2x", rgb2x_glsl},
+	        {"rgb3x", rgb3x_glsl},
+	        {"scan2x", scan2x_glsl},
+	        {"scan3x", scan3x_glsl},
+	        {"sharp", sharp_glsl},
+	        {"tv2x", tv2x_glsl},
+	        {"tv3x", tv3x_glsl},
+	};
+
 	char* src;
 	std::stringstream buf;
 	std::ifstream fshader(shader_path.c_str(), std::ios_base::binary);
-	if (!fshader.is_open()) fshader.open((shader_path + ".glsl").c_str(), std::ios_base::binary);
+	if (!fshader.is_open())
+		fshader.open(shader_path + ".glsl", std::ios_base::binary);
 	if (fshader.is_open()) {
 		buf << fshader.rdbuf();
 		fshader.close();
+	} else if (builtin_shaders.count(shader_path) > 0) {
+		buf << builtin_shaders.at(shader_path);
```

Comment:
When I was testing shaders, I kept checking the console to see confirmation if the shader was loaded, and (after taking screenshots), I looked back to see which shader I was using.. but we're not logging anything.

Would be nice to add a one-liner output on success and the name of the active shader.

Reply:
I believe a log is printed when the name is incorrect / shader file cannot be found. I'll add info log as well :)